modules/tensor_mechanics/include/materials/ADComputeIncrementalShellStrain.h

```diff
@@ -0,0 +1,210 @@
+
+  /// Strain increment in the covariant coordinate system
+  std::vector<ADMaterialProperty(RankTwoTensor) *> _strain_increment;
+
+  /// Total strain increment in the covariant coordinate system
+  std::vector<ADMaterialProperty(RankTwoTensor) *> _total_strain;
+
+  /// Old total strain increment in the covariant coordinate system
+  std::vector<const MaterialProperty<RankTwoTensor> *> _total_strain_old;
+
+  /// Reference to the nonlinear system object
+  NonlinearSystemBase & _nonlinear_sys;
+
+  /// Indices of solution vector corresponding to displacement DOFs in 3 directions at the 4 nodes
+  std::vector<std::vector<unsigned int>> _soln_disp_index;
+
+  /// Indices of solution vector corresponding to rotation DOFs in 2 directions at the 4 nodes
+  std::vector<std::vector<unsigned int>> _soln_rot_index;
+
+  /// Vector that stores the incremental solution at all the 20 DOFs in the 4 noded element.
+  ADDenseVector _soln_vector;
+
+  ADDenseVector _strain_vector;
+
+  std::vector<const Node *> _nodes;
+
+  /// Material property storing the normal to the element at the 4 nodes. Stored as a material property for convinience.
+  ADMaterialProperty(RealVectorValue) & _node_normal;
+
+  /// Material property storing the old normal to the element at the 4 nodes.
+  const MaterialProperty<RealVectorValue> & _node_normal_old;
+
+  /// Quadrature rule in the out of plane direction
+  std::unique_ptr<QGauss> _t_qrule;
+
+  /// Quadrature points in the out of plane direction in isoparametric coordinate system
+  std::vector<Point> _t_points;
+
+  /// Quadrature points in the in-plane direction in isoparametric coordinate system
+  std::vector<Point> _2d_points;
+
+  /// Derivatives of shape functions w.r.t isoparametric coordinates xi
+  std::vector<std::vector<Real>> _dphidxi_map;
+
+  /// Derivatives of shape functions w.r.t isoparametric coordinates eta
+  std::vector<std::vector<Real>> _dphideta_map;
+
+  /// Shape function value
+  std::vector<std::vector<Real>> _phi_map;
+
+  /// Derivative of global x, y and z w.r.t isoparametric coordinate xi
+  std::vector<ADMaterialProperty(RealVectorValue) *> _dxyz_dxi;
+
+  /// Derivative of global x, y and z w.r.t isoparametric coordinate eta
+  std::vector<ADMaterialProperty(RealVectorValue) *> _dxyz_deta;
+
+  /// Derivative of global x, y and z w.r.t isoparametric coordinate zeta
+  std::vector<ADMaterialProperty(RealVectorValue) *> _dxyz_dzeta;
+
+  /// Old derivative of global x, y and z w.r.t isoparametric coordinate xi
+  std::vector<const MaterialProperty<RealVectorValue> *> _dxyz_dxi_old;
+
+  /// Old derivative of global x, y and z w.r.t isoparametric coordinate eta
+  std::vector<const MaterialProperty<RealVectorValue> *> _dxyz_deta_old;
+
+  /// Old derivative of global x, y and z w.r.t isoparametric coordinate zeta
+  std::vector<const MaterialProperty<RealVectorValue> *> _dxyz_dzeta_old;
+
+  /// First tangential vectors at nodes
+  std::vector<ADRealVectorValue> _V1;
```

Comment:
I'd prefer to keep with the MOOSE naming convention and keep variables all lower case. Sometimes, we make exceptions for matrices. Seems like these _V1 and _V2 vectors could be _v1 and _v2